src/vs/workbench/contrib/interactive/browser/interactiveEditor.ts

```diff
@@ -511,107 +510,26 @@ export class InteractiveEditor extends EditorPane {
 		}
 	}
 
-	#lastCell: ICellViewModel | undefined = undefined;
-	#lastCellDisposable = new DisposableStore();
-	#state: ScrollingState = ScrollingState.Initial;
-
-	#cellAtBottom(widget: IActiveNotebookEditorDelegate, cell: ICellViewModel): boolean {
-		const visibleRanges = widget.visibleRanges;
-		const cellIndex = widget.getCellIndex(cell);
-		if (cellIndex === Math.max(...visibleRanges.map(range => range.end))) {
+	#cellAtBottom(cell: ICellViewModel): boolean {
+		const visibleRanges = this.#notebookWidget.value?.visibleRanges || [];
+		const cellIndex = this.#notebookWidget.value?.getCellIndex(cell);
+		if (cellIndex === Math.max(...visibleRanges.map(range => range.end - 1))) {
 			return true;
 		}
 		return false;
 	}
 
-	/**
-	 * - Init state: 0
-	 * - Will cell insertion: check if the last cell is at the bottom, false, stay 0
-	 * 						if true, state 1 (ready for auto reveal)
-	 * - receive a scroll event (scroll even already happened). If the last cell is at bottom, false, 0, true, state 1
-	 * - height change of the last cell, if state 0, do nothing, if state 1, scroll the last cell fully into view
-	 */
-	#registerExecutionScrollListener(widget: NotebookEditorWidget & IActiveNotebookEditorDelegate) {
```

Comment:
We don't need this tracking anymore as the resize and execute event are all that's necessary.

Reply:
:clap: